### PR TITLE
[feature][cpp] Expose Chunking configuration C API

### PR DESCRIPTION
**Pull request by RobertIndie**
### Motivation

Expose Chunking configuration C API

### Modifications

* Expose `chunking_enabled`, `max_pending_chunked_message `, `auto_ack_oldest_chunked_message_on_queue_full ` to C API.
* Add unit tests for C API of producer configuration and consumer configuration.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: pulsar-client-cpp/lib/c/c_ConsumerConfiguration.cc [hunk 1]
```diff
@@ -195,3 +195,25 @@ int pulsar_consumer_configuration_get_priority_level(
...
+}
+
+void pulsar_consumer_configuration_set_auto_ack_oldest_chunked_message_on_queue_full(
+    pulsar_consumer_configuration_t *consumer_configuration,
+    int auto_ack_oldest_chunked_message_on_queue_full) {
+    consumer_configuration->consumerConfiguration.setAutoAckOldestChunkedMessageOnQueueFull(
+        auto_ack_oldest_chunked_message_on_queue_full);
+}
+
+int pulsar_consumer_configuration_get_auto_ack_oldest_chunked_message_on_queue_full(
```
Comment (Demogorgon314) [inline pulsar-client-cpp/lib/c/c_ConsumerConfiguration.cc:216]: Should be name as `pulsar_consumer_configuration_is_auto_ack_oldest_chunked_message_on_queue_full ` ?

Comment (BewareMyPower) [reply pulsar-client-cpp/lib/c/c_ConsumerConfiguration.cc:216]: +1

**Review by shibd — APPROVED** [review]
/LGTM

File: pulsar-client-cpp/tests/ConsumerConfigurationTest.cc [hunk 2]
```diff
@@ -153,6 +154,20 @@ TEST(ConsumerConfigurationTest, testCustomConfig) {
     ASSERT_TRUE(conf.isAutoAckOldestChunkedMessageOnQueueFull());
 }
 
+TEST(ConsumerConfigurationTest, testCApiConfig) {
```
Comment (shibd) [inline pulsar-client-cpp/tests/ConsumerConfigurationTest.cc:157]: Maybe we can open a new folder for the tests for c lib?

**Review by Demogorgon314 — APPROVED** [review]

File: pulsar-client-cpp/tests/c/c_ProducerConfigurationTest.cc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+#include <pulsar/c/producer_configuration.h>
+#include <gtest/gtest.h>
+
+TEST(C_ProducerConfigurationTest, testCApiConfig) {
+    pulsar_producer_configuration_t *producer_conf = pulsar_producer_configuration_create();
+
+    ASSERT_EQ(pulsar_producer_configuration_is_chunking_enabled(producer_conf), false);
+    pulsar_producer_configuration_set_chunking_enabled(producer_conf, true);
+    ASSERT_EQ(pulsar_producer_configuration_is_chunking_enabled(producer_conf), true);
```
Comment (BewareMyPower) [inline pulsar-client-cpp/tests/c/c_ProducerConfigurationTest.cc:27]:
```suggestion
    ASSERT_EQ(pulsar_producer_configuration_is_chunking_enabled(producer_conf), 0);
    pulsar_producer_configuration_set_chunking_enabled(producer_conf, 1);
    ASSERT_EQ(pulsar_producer_configuration_is_chunking_enabled(producer_conf), 1);
```

File: pulsar-client-cpp/tests/c/c_ConsumerConfigurationTest.cc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    ASSERT_EQ(pulsar_consumer_configuration_get_max_pending_chunked_message(consumer_conf), 10);
+    pulsar_consumer_configuration_set_max_pending_chunked_message(consumer_conf, 100);
+    ASSERT_EQ(pulsar_consumer_configuration_get_max_pending_chunked_message(consumer_conf), 100);
+
+    ASSERT_EQ(pulsar_consumer_configuration_is_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf),
+              false);
+    pulsar_consumer_configuration_set_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf, true);
+    ASSERT_EQ(pulsar_consumer_configuration_is_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf),
+              true);
```
Comment (BewareMyPower) [inline pulsar-client-cpp/tests/c/c_ConsumerConfigurationTest.cc:33]:
```suggestion
    ASSERT_EQ(pulsar_consumer_configuration_is_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf),
              0);
    pulsar_consumer_configuration_set_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf, 1);
    ASSERT_EQ(pulsar_consumer_configuration_is_auto_ack_oldest_chunked_message_on_queue_full(consumer_conf),
              1);
```

File: pulsar-client-cpp/tests/ConsumerConfigurationTest.cc [hunk 5]
```diff
@@ -17,6 +17,7 @@
  * under the License.
  */
 #include <pulsar/Client.h>
+#include <pulsar/c/consumer_configuration.h>
```
Comment (BewareMyPower) [inline pulsar-client-cpp/tests/ConsumerConfigurationTest.cc:20]:
```suggestion
```

File: pulsar-client-cpp/tests/ProducerConfigurationTest.cc [hunk 6]
```diff
@@ -18,6 +18,7 @@
  */
 #include <gtest/gtest.h>
 #include <pulsar/ProducerConfiguration.h>
+#include <pulsar/c/producer_configuration.h>
```
Comment (BewareMyPower) [inline pulsar-client-cpp/tests/ProducerConfigurationTest.cc:21]:
```suggestion
```

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
It's better not to use C++ specific semantics in tests for C API. The C interfaces use `int` as a boolean value so we'd better use 1 and 0 to represent true and false.

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**